pkg/aggregator/aggregator.go

```diff
@@ -607,6 +645,33 @@ func (agg *BufferedAggregator) GetEvents() metrics.Events {
 	return events
 }
 
+// EventPlatformDebugEvent contains a single event platform event submitted to the aggregator. If the RawEvent
+// is JSON then it is unmarshalled into UnmarshalledEvent.
+type EventPlatformDebugEvent struct {
+	RawEvent          string `json:",omitempty"`
+	EventType         string
+	UnmarshalledEvent map[string]interface{} `json:",omitempty"`
+}
+
+// GetEventPlatformEvents grabs the event platform events from the queue and clears them.
+// If tryUnmarshalJSON then the raw events will be unmarshalled into JSON if possible before being returned.
+func (agg *BufferedAggregator) GetEventPlatformEvents(tryUnmarshalJSON bool) []EventPlatformDebugEvent {
+	agg.mu.Lock()
+	defer agg.mu.Unlock()
+	events := agg.eventPlatformEvents
+	agg.eventPlatformEvents = nil
+	for i, e := range events {
+		if tryUnmarshalJSON {
+			err := json.Unmarshal([]byte(e.RawEvent), &e.UnmarshalledEvent)
+			if err == nil {
+				e.RawEvent = ""
+			}
+		}
```

Comment:
This seems to be specific for the `check` command and might be better moved there. WDYT ?

Reply:
Good point, moved